### PR TITLE
WIP Torch binary feature

### PR DESCRIPTION
**Pull request by skanjila**
# Code Pull Requests

- converting all datatypes to torch equivalents
- converted tensorflow tensors to torch tensors including adding new dimension at the end of the tensor
- added changes to InputFeature to return torch.Tensor

### CODE REVIEW

File: ludwig/features/base_feature.py [hunk 1]
```diff
@@ -82,9 +82,11 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def create_input(self):
-        return tf.keras.Input(shape=self.get_input_shape(),
-                              dtype=self.get_input_dtype(),
-                              name=self.name + '_input')
+        # return tf.keras.Input(shape=self.get_input_shape(),
+        #                      dtype=self.get_input_dtype(),
+        #                      name=self.name + '_input')
+        return torch.Tensor(shape=self.get_input_shape(),
```
Comment (tgaddair) [inline ludwig/features/base_feature.py:88]: We probably don't need to modify this stuff for now. This isn't needed for PyTorch.

File: ludwig/features/binary_feature.py [hunk 2]
```diff
@@ -181,14 +182,17 @@ def logits(self, inputs, **kwargs):  # hidden
     def predictions(self, inputs, **kwargs):  # hidden
         logits = inputs[LOGITS]
 
-        probabilities = tf.nn.sigmoid(
+        probabilities = torch.nn.sigmoid(
```
Comment (tgaddair) [inline ludwig/features/binary_feature.py:185]: This should be `torch.sigmoid`

File: ludwig/features/binary_feature.py [hunk 3]
```diff
@@ -181,14 +182,17 @@ def logits(self, inputs, **kwargs):  # hidden
...
-            self.threshold,
-            name="predictions_{}".format(self.name),
-        )
+        # predictions = tf.greater_equal(
+        #    probabilities,
+        #    self.threshold,
+        #    name="predictions_{}".format(self.name),
+        # )
+
+        predictions = probabilities > self.threshold
```
Comment (tgaddair) [inline ludwig/features/binary_feature.py:194]: `>=`

File: requirements.txt [hunk 4]
```diff
@@ -7,7 +7,7 @@ tabulate>=0.7
 scikit-learn
 tqdm
 tensorflow>=2.3.1
-tensorflow-addons>=0.13.0
+tensorflow-addons==0.13.0
```
Comment (tgaddair) [inline requirements.txt:10]: Remove tf addons.

Comment (skanjila) [reply requirements.txt]: removed

File: ludwig/features/binary_feature.py (hunk 3 above)
Comment (skanjila) [reply ludwig/features/binary_feature.py]: added

File: ludwig/features/binary_feature.py (hunk 2 above)
Comment (skanjila) [reply ludwig/features/binary_feature.py]: changed

File: ludwig/features/base_feature.py (hunk 1 above)
Comment (skanjila) [reply ludwig/features/base_feature.py]: put the old code back
